phosphor-auditlog/alog_parser.cpp

```diff
@@ -0,0 +1,427 @@
+        {
+            lg2::debug("Unexpected field:{FIELDIDX}", "FIELDIDX", fieldIdx);
+            continue;
+        }
+
+        /* Map the field to the message arg, not all fields are args */
+        mapEntry = msgArgMap.find(fieldName);
+        if (mapEntry != msgArgMap.end())
+        {
+            /* Remove '"' from fieldTxt */
+            messageArgs[mapEntry->second] = getValue(fieldTxt);
+            nFields++;
+#ifdef AUDITLOG_FULL_DEBUG
+            lg2::debug(
+                "Field {NFIELD} : {FIELDNAME} = {FIELDSTR} argIdx = {ARGIDX}",
+                "NFIELD", fieldIdx, "FIELDNAME", fieldName, "FIELDSTR",
+                fieldTxt.c_str(), "ARGIDX", mapEntry->second);
+#endif // AUDITLOG_FULL_DEBUG
+        }
+    } while (auparse_next_field(au) == 1);
+
+    /* Error handling, make sure all the fields we care about
+     * exist. If any are missing set to null string.
+     */
+    if (nFields != msgArgMap.size())
+    {
+#ifdef AUDITLOG_FULL_DEBUG
+        lg2::debug("Incorrect nFields = {NFIELDS}", "NFIELDS", nFields);
+#endif // AUDITLOG_FULL_DEBUG
+
+        // Set missing fields to empty string
+        size_t argIdx;
+        for (argIdx = 0;
+             (nFields != msgArgMap.size()) && (argIdx < msgArgMap.size());
+             argIdx++)
+        {
+            if (messageArgs[argIdx] == nullptr)
+            {
+#ifdef AUDITLOG_FULL_DEBUG
+                lg2::debug("Correcting arg={ARGIDX}", "ARGIDX", argIdx);
+#endif // AUDITLOG_FULL_DEBUG
+                messageArgs[argIdx] = "";
+                nFields++;
+            }
+        }
+
+#ifdef AUDITLOG_FULL_DEBUG
+        lg2::debug("nFields = {NFIELDS}", "NFIELDS", nFields);
+#endif // AUDITLOG_FULL_DEBUG
+    }
+
+    parsedEntry["MessageArgs"] = std::move(messageArgs);
+}
+
+bool ALParser::formatMsgReg(nlohmann::json& parsedEntry)
+{
+    /* Fill common fields for any record type */
+    auto fullTimestamp = auparse_get_timestamp(au);
+    if (fullTimestamp == nullptr)
+    {
+        lg2::error("Failed to parse timestamp");
+        throw sdbusplus::xyz::openbmc_project::Common::Error::InternalFailure();
+        return false;
+    }
+    parsedEntry["EventTimestamp"] = fullTimestamp->sec;
+    parsedEntry["ID"] = std::format("{}.{}:{}", fullTimestamp->sec,
+                                    fullTimestamp->milli,
+                                    fullTimestamp->serial);
+
+    /* Fill varied args fields based on record type */
+    int recType = auparse_get_type(au);
+
+    switch (recType)
+    {
+        case AUDIT_USYS_CONFIG:
+            fillUsysEntry(parsedEntry);
+            break;
+
+        default:
+            fillAuditEntry(parsedEntry);
+            break;
+    }
+
+#ifdef AUDITLOG_FULL_DEBUG
+    lg2::debug("parsedEntry = {PARSEDENTRY}", "PARSEDENTRY",
+               parsedEntry.dump());
+#endif // AUDITLOG_FULL_DEBUG
+
+    return true;
+}
+
+bool ALParser::formatGen(nlohmann::json& parsedEntry)
```

Comment:
maybe use 'formatGeneral' here, because to me formatGen sounds like formatGenerate

Reply:
Good suggestion, I've changed it.